client/src/main/proto/app/cash/backfila/client_service.proto

```diff
@@ -95,17 +95,23 @@ message RunBatchRequest {
   // User provided parameters at creation.
   map<string, bytes> parameters = 5;
   optional bool dry_run = 6;
-
-  // Provided data from the source backfill if this is the target backfill in a pipelined backfill.
-  optional PipelinedData pipelined_data = 7;
 }
 message RunBatchResponse {
   // How much to sleep before starting new batches. For example can be set to replication lag.
   optional uint64 backoff_ms = 1;
 
   optional PipelinedData pipelined_data = 2;
 
+  // Stacktrace that will be surfaced to the user.
+  // If this is provided this batch request is assumed to have failed.
   optional string exception_stack_trace = 3;
+
+  // Provided data from the backfila client telling backfila what part of the batch was not
+  // completed yet. Backfila will end up retrying retry either a new request with the remaining
```

Comment:
retrying retry ---> I think you want to say “resuming” not retrying?

Reply:
Backfila will end up resuming with a new request containing either the remaining range or the original range.